settings/acceptable_versions.go

```diff
@@ -0,0 +1,146 @@
+package settings
+
+import (
+	"fmt"
+	"github.com/packwiz/packwiz/cmdshared"
+	"github.com/packwiz/packwiz/core"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/unascribed/FlexVer/go/flexver"
+	"golang.org/x/exp/slices"
+	"os"
+	"strings"
+)
+
+var acceptableVersionsCommand = &cobra.Command{
+	Use:     "acceptable-versions",
+	Short:   "Manage your pack's acceptable Minecraft versions. This must be a comma seperated list of Minecraft versions, e.g. 1.16.3,1.16.4,1.16.5",
+	Aliases: []string{"av"},
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		modpack, err := core.LoadPack()
+		if err != nil {
+			// Check if it's a no such file or directory error
+			if os.IsNotExist(err) {
+				fmt.Println("No pack.toml file found, run 'packwiz init' to create one!")
+				os.Exit(1)
+			}
+			fmt.Printf("Error loading pack: %s\n", err)
+			os.Exit(1)
+		}
+		var currentVersions []string
+		// Check if they have no options whatsoever
+		if modpack.Options == nil {
+			// Initialize the options
+			modpack.Options = make(map[string]interface{})
+		}
+		// Check if the acceptable-game-versions is nil, which would mean their pack.toml doesn't have it set yet
+		if modpack.Options["acceptable-game-versions"] != nil {
+			// Convert the interface{} to a string slice
+			for _, v := range modpack.Options["acceptable-game-versions"].([]interface{}) {
+				currentVersions = append(currentVersions, v.(string))
+			}
+		}
+		// Check our flags to see if we're adding or removing
+		if viper.GetBool("settings.acceptable-versions.add") {
+			// Adding
+			acceptableVersion := args[0]
+			// Check if the version is already in the list
+			if slices.Contains(currentVersions, acceptableVersion) {
+				fmt.Printf("Version %s is already in your acceptable versions list!\n", acceptableVersion)
+				os.Exit(1)
+			}
+			// Add the version to the list and re-sort it
+			currentVersions = append(currentVersions, acceptableVersion)
+			flexver.VersionSlice(currentVersions).Sort()
+			// Set the new list
+			modpack.Options["acceptable-game-versions"] = currentVersions
+			// Save the pack
+			err = modpack.Write()
+			if err != nil {
+				fmt.Printf("Error writing pack: %s\n", err)
+				os.Exit(1)
+			}
+			// Print success message
+			prettyList := strings.Join(currentVersions, ", ")
+			fmt.Printf("Added %s to acceptable versions list, now %s\n", acceptableVersion, prettyList)
+		} else if viper.GetBool("settings.acceptable-versions.remove") {
+			// Removing
+			acceptableVersion := args[0]
+			// Check if the version is in the list
+			if !slices.Contains(currentVersions, acceptableVersion) {
+				fmt.Printf("Version %s is not in your acceptable versions list!\n", acceptableVersion)
+				os.Exit(1)
+			}
+			// Remove the version from the list
+			currentVersions = slices.Delete(currentVersions, slices.Index(currentVersions, acceptableVersion), 1)
```

Comment:
The third argument to `slices.Delete` is the end index (exclusive) not the length to remove! I've already checked it out, so I'll just fix it myself :P